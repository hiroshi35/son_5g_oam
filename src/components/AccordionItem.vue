<template>
  <div class="panel accordion">
    <div class="panel-heading" role="tab" :id="'heading_' + id">
      <h4 class="panel-title">
        <a role="button" data-toggle="collapse" :data-parent="'#' + parentId" :href="'#' + id" aria-expanded="true"
           :aria-controls="id" v-html="title">
        </a>
      </h4>
    </div>
    <div :id="id"
         :class="{'panel-collapse collapse in': active, 'panel-collapse collapse': !active }"
         role="tabpanel" :aria-labelledby="'heading_' + id">
      <div class="panel-body">
        <slot></slot>
      </div>
    </div>
  </div>
</template>

<script>
  export default {
    props: ['title', 'active'],
    computed: {
      id () {
        return this.title.replace(/[^a-zA-Z0-9]+/g, '_').toLowerCase()
      },
      parentId () {
        return this.$parent.id
      }
    }
  }
</script>