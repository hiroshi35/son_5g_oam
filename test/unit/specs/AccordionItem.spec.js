import { shallow } from 'vue-test-utils'
import AccordionItem from '@/components/AccordionItem.vue'

describe('AccordionItem.vue', () => {
  let wrapper
  AccordionItem.beforeCreate = function () { this.$parent = {id: 'accordion_id'} }
  beforeEach(() => {
    wrapper = shallow(AccordionItem,
      {
        propsData: {
          title: 'Accordion'
        }
      })
  })
  it('should be vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })
  it('has the expected html structure', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it('has correct props', () => {
    expect(wrapper.props().title).toBe('Accordion')
  })
})
