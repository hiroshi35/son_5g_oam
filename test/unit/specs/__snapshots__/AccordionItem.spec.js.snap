// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordionItem.vue has the expected html structure 1`] = `
<div
  class="panel accordion"
>
  <div
    class="panel-heading"
    id="heading_accordion"
    role="tab"
  >
    <h4
      class="panel-title"
    >
      <a
        aria-controls="accordion"
        aria-expanded="true"
        data-parent="#accordion_id"
        data-toggle="collapse"
        href="#accordion"
        role="button"
      >
        Accordion
      </a>
    </h4>
  </div>
   
  <div
    aria-labelledby="heading_accordion"
    class="panel-collapse collapse"
    id="accordion"
    role="tabpanel"
  >
    <div
      class="panel-body"
    />
  </div>
</div>
`;
